### PR TITLE
chore(ci): increase release automation and limit to one manual step

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue number:** #1009 

### Changes

> Please provide a summary of what's being changed

This PR reduces operational overhead when releasing a new version by restricting manual steps to a single job: Drafting release notes. 

**Main changes**

* Changelog generation and version bump are now part of the release process
* Release process can be triggered manually in case of GH Actions intermittent failures with builds to start
* Split release jobs into three to ease further maintenance: 1/ release PyPi/SAR, 2/ docs, 3/ post release

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of my this change
* [x] Changes are tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

**Tasks**

- [x] Remove version checking
- [x] Push staged files pyproject/changelog to trunk
- [x] Prevent concurrent docs pushing different changelog versions
- [x] Split docs and auto-close issues
- [x] Update manual x automated activities step
- [x] Split jobs to ease further maintenance
  - [x] Make sure git is configured in separate job just in case
- [x] Update failure mechanism
- [x] Update MAINTAINERS.md release steps
- [x] Include manual dispatch for failures

**Separate PR to limit failure radius**

- [ ] Make docs build workflow reusable
- [ ] Include changelog bump on every PR merge (unreleased section)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

-----
[View rendered .chglog/CHANGELOG.tpl.md](https://github.com/heitorlessa/aws-lambda-powertools-python/blob/chore/changelog-automate/.chglog/CHANGELOG.tpl.md)
[View rendered MAINTAINERS.md](https://github.com/heitorlessa/aws-lambda-powertools-python/blob/chore/changelog-automate/MAINTAINERS.md)

### CODE REVIEW

**Comment by heitorlessa** [comment]
@rubenfonseca @leandrodamascena -- need your input the most here. Please let me know if this is clear, and whether there are any areas you want to me further document now (or later?).

File: .github/workflows/publish.yml
Comment (leandrodamascena) [inline .github/workflows/publish.yml:18]: Great move removing many manual steps and making it automatic. It reduces many human interactions and avoids manual errors.

File: .github/workflows/publish.yml
Comment (leandrodamascena) [inline .github/workflows/publish.yml:38]: Should this get a default value if the tag name is not present? We know that a tag will always be present, but I think you should consider testing if the tag name/value is not present for some reason.

**Comment by leandrodamascena** [comment]
Hello @heitorlessa. Thanks for tagging me.
I added two comments but this PR looks amazing. This will reduce friction in the manual steps and make things more automated.

This PR makes announcing new releases even more interesting as it automates all the work in the repository and allows for human interaction by writing text (and with images) announcing the release.

I can't think of anything to add to the documentation right now. But I'll let you know if I think of anything new. 

Congrats :100: :rocket:

File: .github/workflows/publish.yml
Comment (heitorlessa) [reply .github/workflows/publish.yml:38]: good catch! I'll add this into our list to handle this later https://github.com/awslabs/aws-lambda-powertools-python/issues/1009. I'd generally solve it now but because of our recent hiccups with GitHub Actions failing conditional expressions, I'd like to make sure this release works first before we add additional defenses if that makes sense. 

Feel free to call me out if I'm being overzealous.

**Comment by heitorlessa** [comment]
@rubenfonseca need your approval before I fix a conflict and merge

File: .github/workflows/publish.yml
Comment (leandrodamascena) [reply .github/workflows/publish.yml:38]: I don't think you're being overzealous and I think it's okay to add it to the Papercut list to prioritize after solving other issues.

To be honest, I believe it will never fail unless GitHub changes its schema. But we never know and if we can add conditional validation, I think it will make this step more resilient.

I can open a PR after the freezing.

Thanks man.

**Review by rubenfonseca — APPROVED** [review]
This is great work. Have you considered writing a book on Github Actions automation? 💯

**Pull request merged**
